addons/mail/static/src/models/message/message.js

```diff
@@ -201,6 +201,22 @@ function factory(dependencies) {
             });
         }
 
+        /**
+         * Removes the message body. Unlinking the record would pose many
+         * problems in places that reference the message.
+         */
+        delete() {
+            const data = {
+                body: "",
+            }
+            this.update(data);
+            return this.async(() => this.env.services.rpc({
+                model: 'mail.message',
+                method: 'write',
+                args: [this.id, data],
+            }));
```

Comment:
Doing it this way means if the server is down the user will never know it wasn't actually updated, but it will look like it was.